### PR TITLE
kube-oidc-proxy: cleanup some certs logic

### PR DESCRIPTION
**Pull request by jieyu**
I happened to notice the issue when I was working on another
improvement. The chart is not really conditional on `tls.enabled` as it
always expect the server certs to be present in the deployment. Thus, I
remove all the conditional logics.

### CODE REVIEW

File: staging/kube-oidc-proxy/templates/certificate.yaml [hunk 1]
```diff
@@ -1,11 +1,10 @@
-{{- if .Values.tls.enabled }}
 apiVersion: certmanager.k8s.io/v1alpha1
 kind: Certificate
 metadata:
   name: kube-oidc-proxy
   namespace: {{ .Release.Namespace }}
 spec:
-  secretName: kube-oidc-proxy-kubeaddons-server-tls
+  secretName: {{ include "kube-oidc-proxy.fullname" . }}-server-tls
```
Comment (mhrabovcin) [inline staging/kube-oidc-proxy/templates/certificate.yaml]: is this backward compatible for existing installations?

File: staging/kube-oidc-proxy/templates/deployment.yaml [hunk 2]
```diff
@@ -1,6 +1,4 @@
 {{ $fullname := include "kube-oidc-proxy.fullname" . }}
-{{ $defaultTlsSecretName := printf "%s-tls" $fullname }}
```
Comment (mhrabovcin) [inline staging/kube-oidc-proxy/templates/deployment.yaml]: 👍

File: staging/kube-oidc-proxy/templates/certificate.yaml (hunk 1 above)
Comment (jieyu) [reply staging/kube-oidc-proxy/templates/certificate.yaml]: @mhrabovcin yes, i tested with an old verison, and perform an `helm upgrade kube-oidc-proxy .` and observed no change.

**Comment by jieyu** [comment]
@mhrabovcin i rebased and need re-approval.

File: staging/kube-oidc-proxy/templates/certificate.yaml (hunk 1 above)
Comment (makkes) [reply staging/kube-oidc-proxy/templates/certificate.yaml]: We need tests to automate backwards compatibility. *sigh*

**Pull request merged**